src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c

```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjBlockstop_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBlockstop* this = THIS;
+    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    this->actionFunc = ObjBlockstop_CheckCollision;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Init.s")
+void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
+    Actor* tempActor;
+    tempActor = globalCtx->actorCtx.actorLists[6].first;
+    while (tempActor) {
```

Comment:
```suggestion
void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
    Actor* tempActor = globalCtx->actorCtx.actorLists[ACTORCAT_PROP].first;
    
    while (tempActor) {
```

Could this be put all on the same line?
Also, you can use the enum value for the `6`.